tests/abi/abi.c

```diff
@@ -0,0 +1,175 @@
+    FOR_EACH(PUSH_REGISTER, REGISTER_LIST)                                   \
+    /* Popularize the stack with random data */                              \
+    REPEAT(PUSH_CONSTANT, STACK_VALUE_COUNT, REGISTER_COUNT, FIRST_REGISTER) \
+    /* Popularize all the registers with random data */                      \
+    INDEXED_FOR_EACH(FILL_REGISTER, REGISTER_LIST)                           \
+    /* Call the function */                                                  \
+    CALL_A_FUNCTION(FUNCTION_NAME)                                           \
+    /* Restore the stack */                                                  \
+    RESTORE_STACK(STACK_VALUE_COUNT)                                         \
+    /* Restore all the registers */                                          \
+    REVERSE_FOR_EACH(POP_REGISTER, REGISTER_LIST)                            \
+    :: "X"(Generated)                                                        \
+    : COMMA_SEPARATED_FOR_EACH(STRINGIFY, REGISTER_LIST), "memory"           \
+  );                                                                         \
+                                                                             \
+  free(Generated);                                                           \
+}
+/* clang-format on */
+
+#ifdef __APPLE__
+#define FUNCTION_NAME_PREFIX _
+#else
+#define FUNCTION_NAME_PREFIX
+#endif
+
+#ifdef REG_
+#undef REG_
+#endif
+#define REG_ "%"
+
+#ifdef LOC_
+#undef LOC_
+#endif
+#define LOC_ "%" STRINGIFY(FRAME_POINTER)
+
+/* clang-format off */
+#define NAKED_FUNCTIONS __asm__ (                                          \
+  /* Define the first function */                                          \
+  STRINGIFY(FUNCTION_NAME_PREFIX) "getTheValue:\n"                         \
+  UPDATE_FRAME_POINTER                                                     \
+  ALLOCATE_ON_HEAP(FRAME_POINTER, (GENERATED_COUNT) * (REGISTER_SIZE))     \
+  CALL_A_FUNCTION_WITH_FRAME_POINTER(generate)                             \
+  CALL_A_FUNCTION_WITH_FRAME_POINTER(dump)                                 \
+                                                                           \
+  FOR_EACH(PUSH_REGISTER, REGISTER_LIST)                                   \
+  REPEAT(PUSH_CONSTANT, STACK_VALUE_COUNT, REGISTER_COUNT, FIRST_REGISTER) \
+  INDEXED_FOR_EACH(FILL_REGISTER, REGISTER_LIST)                           \
+                                                                           \
+  FIX_RETURN_ADDRESS((GENERATED_COUNT + 1) * (REGISTER_SIZE))              \
+  RETURN_FROM_FUNCTION                                                     \
+                                                                           \
+  /* Define the second function */                                         \
+  STRINGIFY(FUNCTION_NAME_PREFIX) "restoreTheState:\n"                     \
+  POP_REGISTER(rbp)                                                        \
```

Comment:
This is not portable

Reply:
Whoops, right, my bad, there's already macro for that (see [here](https://github.com/revng/revng-qa/pull/4/files#diff-86d497e9de7e66dcdab3f18d737c7e11b83568d711458596246551f26e75d5e6R135)). That would have been easily caught as soon as I added one more supported architecture. But thanks for pointing it out, I'll fix it.

Reply:
Gone.